### PR TITLE
#3530 enhance sanity testing framework

### PR DESCRIPTION
**Pull request by samuelcostae**
### Description

Enhanced Sanity  Tests 

### Issues Resolved

#3530

### Check List

- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by samuelcostae** [comment]
@DarshitChanpura  the tests are basically done, but based on the clarification on the issue discussion, i still have to replace the actual calls to the backend with mock responses.

I'm moving this out of draft without having complete that step I can get any feedback on the overall structure of the tests

File: cypress/integration/plugins/security-dashboards-plugin/sanity_tests.spec.js
Comment (SuZhou-Joe) [inline cypress/integration/plugins/security-dashboards-plugin/sanity_tests.spec.js:12]: Is this package needed?

File: cypress/integration/plugins/security-dashboards-plugin/sanity_tests.spec.js
Comment (SuZhou-Joe) [inline cypress/integration/plugins/security-dashboards-plugin/sanity_tests.spec.js:156]: Please use BASE_PATH instead of a hard-coded host:port.

File: cypress/integration/plugins/security-dashboards-plugin/sanity_tests.spec.js
Comment (SuZhou-Joe) [inline cypress/integration/plugins/security-dashboards-plugin/sanity_tests.spec.js:70]: Maybe a risk if OSD is started with prefix enabled. Recommend using BATH_PATH.

**Comment by samuelcostae** [comment]
@DarshitChanpura  friendly ping

**Review by DarshitChanpura — APPROVED** [review]
LGTM

**Comment by DarshitChanpura** [comment]
Could we get some maintainer reviews on this?

In the latest CI run, cy.wait didn't seem to have caused any issues

**Comment by ruanyl** [comment]
@samuelcostae should this backport to `2.x`?

**Comment by DarshitChanpura** [comment]
> @samuelcostae should this backport to 2.x?

yes.

**Comment by DarshitChanpura** [comment]
@SuZhou-Joe @xluo-aws @Hailong-am @ruanyl @wanglam @raintygao Can we get some reviews on this PR? Last update was more than a month ago.

File: cypress/integration/plugins/security-dashboards-plugin/sanity_tests.spec.js
Comment (Hailong-am) [inline cypress/integration/plugins/security-dashboards-plugin/sanity_tests.spec.js:198]: this can be simplified by https://github.com/opensearch-project/opensearch-dashboards-functional-test/blob/main/cypress/utils/commands.js#L526

File: cypress/integration/plugins/security-dashboards-plugin/sanity_tests.spec.js
Comment (Hailong-am) [inline cypress/integration/plugins/security-dashboards-plugin/sanity_tests.spec.js:184]: do we need any validation for newly created index pattern?

Comment (samuelcostae) [reply cypress/integration/plugins/security-dashboards-plugin/sanity_tests.spec.js:182]: I've  added a simple check to see if the list of index patterns include the created index. Let me know if that requires further changes

File: cypress/integration/plugins/security-dashboards-plugin/sanity_tests.spec.js
Comment (samuelcostae) [reply cypress/integration/plugins/security-dashboards-plugin/sanity_tests.spec.js]: I've replaced it with the command

**Review by Hailong-am — APPROVED** [review]

**Comment by Hailong-am** [comment]
@samuelcostae would you mind past local run result? The CI flow not running for the security-dashbaord-plugin and create an issue to track https://github.com/opensearch-project/opensearch-dashboards-functional-test/issues/1139

**Comment by samuelcostae** [comment]
> @samuelcostae would you mind past local run result? The CI flow not running for the security-dashbaord-plugin and create an issue to track #1139

**Comment by DarshitChanpura** [comment]
@SuZhou-Joe @xluo-aws @ruanyl @wanglam @raintygao Can we get 1 more review on this PR?

**Comment by DarshitChanpura** [comment]
@SuZhou-Joe @xluo-aws @ruanyl @wanglam @raintygao Can we get 1 more review on this PR please?

**Comment by DarshitChanpura** [comment]
@davidlago @wbeckler can you please get this unblocked?

**Comment by samuelcostae** [comment]
> @davidlago @wbeckler can you please help get this unblocked?

@davidlago @wbeckler Just a friendly ping for a second review.

Let us know any further work is required.

**Comment by DarshitChanpura** [comment]
@SuZhou-Joe @xluo-aws @ruanyl @wanglam @raintygao Pinging for 1 more review on this PR please. This has been lingering for more than 2 weeks now.

**Review by ruanyl — APPROVED** [review]

**Comment by kavilla** [comment]
One comment is that we should consider on cleaning up the addition of the sample data within the test itself. Otherwise it leaves the cluster in a dirty state and could case the next test to fail.

I would also verify that the expected test index isn't already existing to avoid transient failures.

**Pull request merged**
